src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -70,25 +70,25 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
                 this.RegisterService<ICSharpTempPECompilerService>(async ct =>
                 {
+                    var workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
                     await JoinableTaskFactory.SwitchToMainThreadAsync(ct);
-                    return new TempPECompilerService(this.Workspace.Services.GetService<IMetadataService>());
+                    return new TempPECompilerService(workspace.Services.GetService<IMetadataService>());
                 });
             }
             catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e))
             {
             }
         }
 
-        protected override VisualStudioWorkspaceImpl CreateWorkspace()
-            => this.ComponentModel.GetService<VisualStudioWorkspaceImpl>();
-
         protected override async Task RegisterObjectBrowserLibraryManagerAsync(CancellationToken cancellationToken)
         {
+            var workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
```

Comment:
could this even move into ObjectBrowserLibraryManager? (since it captures a reference to both this and ComponentModel?

Reply:
Yeah, possibly; I think there's potentially bigger work there too to not even load the component until you're actually using the object browser...

Reply:
(sorry didn't see the comment until after hitting merge)